pubtools/_quay/operator_pusher.py

```diff
@@ -420,30 +429,64 @@ def build_index_images(self):
         iib_results = {}
 
         for version, items in sorted(self.version_items_mapping.items()):
-            bundles = [self.public_bundle_ref(i) for i in items]
-            all_archs = [
-                i.metadata["arch"] if i.metadata["arch"] != "x86_64" else "amd64" for i in items
-            ]
-            archs = sorted(list(set(all_archs)))
-            signing_keys = sorted(list(set([item.claims_signing_key for item in items])))
+            is_hotfix = any([item.metadata.get("com.redhat.hotfix") for item in items])
+            is_advisory_source = all(
+                [re.match(r"^[A-Z0-9:\-]{4,40}$", item.origin) for item in items]
+            )
+            print([item.metadata.get("com.redhat.hotfix") for item in items])
+            print([item.origin for item in items])
+            item_groups = {}
+            if is_hotfix and not is_advisory_source:
+                raise ValueError("Cannot push hotfixes without an advisory")
+            if is_hotfix:
+                for item in items:
+                    item_groups.setdefault(item.origin, []).append(item)
+            else:
+                item_groups["default"] = items
+
+            for group, g_items in item_groups.items():
+                if is_hotfix:
+                    tag = "{0}-{1}-{2}".format(
+                        version,
+                        items[0].metadata["com.redhat.hotfix"],
+                        items[0].origin.replace(":", "-"),
```

Comment:
Above you put items to group based on different origins, but here you consider every origin to be the same?